toolchain/check/eval_inst.cpp

```diff
@@ -348,13 +349,30 @@ auto EvalConstantInst(Context& context, SemIRLoc loc,
               interface_fn_args.end());
   auto specific_id = MakeSpecific(context, loc, generic_id, args);
 
-  // TODO: Add the new `SpecificFunction` to definitions_required.
+  if (!SemIR::GetCalleeFunction(context.sem_ir(), inst.callee_id)
+           .self_type_id.has_value()) {
```

Comment:
I think we don't need this check in this case -- we just fetched the function from an impl witness table, so it can't be a function declaration from an interface.